### PR TITLE
Add generate QR code

### PR DESCRIPTION
**Pull request by wyhaya**
preview

#45

### CODE REVIEW

**Comment by svenstaro** [comment]
Also this needs a rebase.

**Comment by svenstaro** [comment]
Do you also want to tackle the task of printing a QR code in the terminal if requested via option flag? That was part of issue #45 but it should likely be done in a separate PR if you want to do it.

**Comment by wyhaya** [comment]
> I think there should be a CLI switch to disable this function (to keep the UI tidy for people who don't like the visual bloat).

It may be considered to add a CLI switch, but is there too many switches at present, it is difficult for users to understand each function, maybe we can make the style smaller?

> Could you add a parametrized test for this that checks multiple values for the qrcode param and see whether you can trigger the error, too?

Ok i will add as soon as possible.

> Do you also want to tackle the task of printing a QR code in the terminal if requested via option flag? That was part of issue #45 but it should likely be done in a separate PR if you want to do it.

Yes, but what characters should we use? The terminal cannot be completely set to a square, because it is impossible to control the user’s ‘line-height’, it is usually difficult to scan successfully

Reference: [qrcode-rust](https://github.com/kennytm/qrcode-rust#string-generation)

**Comment by svenstaro** [comment]
> > I think there should be a CLI switch to disable this function (to keep the UI tidy for people who don't like the visual bloat).
> 
> It may be considered to add a CLI switch, but is there too many switches at present, it is difficult for users to understand each function, maybe we can make the style smaller?

I definitely am worried about making the tool too complex as it touts itself as being simple. However, I think at current we can handle a `--no-qr` (or similar) flag just fine.

> 
> > Do you also want to tackle the task of printing a QR code in the terminal if requested via option flag? That was part of issue #45 but it should likely be done in a separate PR if you want to do it.
> 
> Yes, but what characters should we use? The terminal cannot be completely set to a square, because it is impossible to control the user’s `line-height`, it is usually difficult to scan successfully
> 
> Reference: [qrcode-rust](https://github.com/kennytm/qrcode-rust#string-generation)

True, though its [unicode mode](https://github.com/kennytm/qrcode-rust#unicode-string-generation) works well enough for me so let's just roll with that for the time being.

**Comment by svenstaro** [comment]
FWIW the unicode string mode is gapless for me.

**Comment by wyhaya** [comment]
I added the `--qrcode` option, which is `false` by default. Do you think it is appropriate?

File: src/renderer.rs [hunk 1]
```diff
@@ -133,24 +134,33 @@ fn color_scheme_selector(
...
     html! {
         nav {
-            ul {
-                li {
-                    a.change-theme href="#" title="Change theme" {
-                        "Change theme..."
+            @if show_qrcode {
+                div {
+                    p onmouseover="document.querySelector('#qrcode').src = `/?qrcode=${encodeURIComponent(window.location.href)}`" {
+                        "QrCode"
```
Comment (svenstaro) [inline src/renderer.rs:144]: Minor nitpick but can you call this `QR code` instead? :)

File: src/renderer.rs [hunk 2]
```diff
@@ -133,24 +134,33 @@ fn color_scheme_selector(
...
-                            @if active_color_scheme == color_scheme {
-                                li.active {
-                                    (color_scheme_link(sort_method, sort_order, color_scheme, default_color_scheme, serve_path))
-                                }
-                            } @else {
-                                li {
-                                    (color_scheme_link(sort_method, sort_order, color_scheme, default_color_scheme, serve_path))
-                                }
+                    div.qrcode {
+                        img#qrcode alt="QrCode" title="QR code of this page";
```
Comment (svenstaro) [inline src/renderer.rs:147]: Also in this alt name as above.

**Pull request merged**